flybywiresim/a32nx
# Add fs-base-ui typedefs

## Summary of Changes
Add TypeScript type definitions for a lot of fs-base-ui types to help us develop correct code quicker and easier.

_Some notes_:

As the Revealing Module Pattern is used a lot in Asobo's code, many of the definitions contain namespaces, which are best suited to represent this pattern.

Given that `SimVar.GetSimVarValue` may return `null`, a lot of the `SimPlane` functions that use `SimVar.GetSimVarValue` may also return `null`. Only functions that cannot return `null` (e.g. because they use `Math.max` which never returns `null`) have an output type without `null`.

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): SjotgunSjonnie#9623


## How to download the PR for QA

**Note that QA doesn't apply to this PR, as the PR doesn't contain any code that runs inside the sim.**

Every new commit to this PR will cause a new A32NX artifact to be created and uploaded.
The build script will have already been run with the latest changes, so no need to rerun it once you download the zip.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the right side, click on the **Artifacts** drop down and click the **A32NX** link
